src/QirRuntime/test/QIR-tracer/tracer-config.hpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// TODO: ideally, this file should be generated by the Q# compiler alongside the qir
+
+#pragma once
+
+#include <string>
+#include <unordered_map>
+
+namespace Microsoft
+{
+namespace Quantum
+{
+    using OpId = int;
+    using Time = int;
+    using Duration = int;
+    using LayerId = size_t;

```

Comment:
I saw this earlier. You may want to extract into a header and include that header where needed.
Otherwise in one place this can be changed, in the other place forgotten to update - discrepancy (and bugs hard to catch).

Reply:
Yes. Moved the typedefs into a header in `public` folder.